antivir/models/scanner.py

```diff
@@ -0,0 +1,169 @@
+        """
+        if not results:
+            results = []
+        return results
+
+    @api.multi
+    def bulk_scan(self, limit):
+        """ Chooses attachments without scan date (not scanned yet) and then runs scan. """
+        config = self.env['antivir.config'].search([
+            ('active_config', '=', True)
+        ])
+
+        if config.exists() and config.scan_cron:
+            attachment_ids = self.env['ir.attachment'].search([('virus_scan_date', '=', False), ('type', '=', 'binary')], limit=limit)
+
+            for attachment in attachment_ids:
+                datas = attachment.datas
+                if datas:
+                    decoded_datas = base64.b64decode(datas)
+                    SHA256 = hashlib.sha256(datas).hexdigest()
+                    result = self.scan(decoded_datas)
+
+                    if any(x.values()[0] for x in result):
+                        threat = ','.join([
+                            val[1] for d in result for key, val in d.iteritems() if val is not None
+                        ])
+
+                        now = datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+                        quarantine_id = self.env['antivir.quarantine'].create({
+                            'name': '{}-{}'.format(attachment.datas_fname, now),
+                            'filename': attachment.datas_fname,
+                            'threat': threat,
+                            'scan_results': result,
+                            'file_data': datas,
+                            'SHA256': SHA256,
+                            'user_id': uid,
+                            'quarantined': now
+                        })
+
+                        attachment.write({
+                            'virus_scan_date': now,
+                            'virus_safe': False,
+                            'virus_found_name': threat,
+                            'quarantine_id': quarantine_id
+                        })
+                        cr.commit()
+                        _logger.info(_("Virus Found in attachment. File added to quarantine."))
+                    else:
+                        now = datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+                        attachment.write({
+                            'virus_scan_date': now,
+                            'virus_safe': True
+                        })
+                else:
+                    now = datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+                    attachment.write({
+                        'virus_scan_date': now,
+                        'virus_safe': True
+                    })
+
+    @contextmanager
+    @api.model
+    def active_scanner(self):
+        is_active = False
+        scanner_ids = self.search([
+            ('name', '=', self._av_engine)
+        ])
+        try:
+            if scanner_ids.exists():
+                is_active = scanner_ids[0].is_active
```

Comment:
yield scanner_ids[0].is_active

Reply:
I agree with you, of course it was a poor migration of mine.